packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,224 @@
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * Disable template validation.
+   * @default True
+   */
+  readonly disableTemplateValidation?: boolean;
+
+  /**
+   * The S3 url that points to the provisioning artifacts template
+   */
+  readonly templateUrl: string;
+
+  /**
+   * The name of the provisioning artifact.
+   * @default - No provisioning artifact name provided
+   */
+  readonly name?: string;
+}
+
+/**
+ * Properties for a Cloudformation Product
+ */
+export interface ProductProps {
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+
+  /**
+   * The language code.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * The description of the product.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * The distributor of the product.
+   * @default - No distributor provided
+   */
+  readonly distributor?: string;
+
+  /**
+   * The owner of the product.
+   */
+  readonly owner: string;
+
+  /**
+   * The configuration of the provisioning artifact (also known as a version).
+   */
+  readonly provisioningArtifacts: ProvisioningArtifactProperties[];
+
+  /**
+   * ReplaceProvisioningArtifacts
+   * @default false
+   */
+  readonly replaceProvisioningParameters?: boolean;
+
+  /**
+   * The support information about the product
+   * @default - No support description provided
+   */
+  readonly supportDescription?: string;
+
+  /**
+   * The contact email for product support.
+   * @default - No support email provided
+   */
+  readonly supportEmail?: string;
+
+  /**
+   * The contact URL for product support.
+   * @default - No support URL provided
+   */
+  readonly supportUrl?: string;
+}
+
+/**
+ * A Service Catalog Cloudformation Product.
+ * @resource AWS::ServiceCatalog::CloudFormationProduct
+ */
+export class Product extends ProductBase {
+  /**
+   * Creates a Product construct that represents an external product.
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs Product import properties
+   */
+  public static fromProductAttributes(scope: Construct, id: string, attrs: ProductAttributes): IProduct {
+    const parts = Stack.of(scope).parseArn(attrs.productArn);
+    function throwError(errorMessage: string): never {
+      throw new Error(errorMessage);
+    }
+    class Import extends ProductBase {
+      public readonly productId = parts.resourceName ?? throwError('Product arn missing Product ID during import from attributes');
+      public readonly productArn = attrs.productArn;
+      public readonly productName = attrs.productName;
+    }
+    return new Import(scope, id);
+  }
+
+  public readonly productArn: string;
+  public readonly productName: string;
+  public readonly productId: string;
+  private readonly product: CfnCloudFormationProduct;
```

Comment:
I don't think this is read anywhere beside the constructor, right? Let's remove this field, and turn it into a local variable.